### ISSUE TITLE
USWDS-Site - Feature: Add additional accessibility and other guidance to Validation component page

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds-site#2481]
### Is your feature request related to a problem? Please describe.

When auditing the [validation component](https://designsystem.digital.gov/components/validation/) for accessibility compliance, it wasn't initially clear that the validation is not a form of error handling.

It also wasn't clear that the only type of input validation can be used for is text input (is it?)

It would help to have a more robust example of how and when to implement the validation component.

### Describe the solution you'd like

I recommend we add more robust guidance and another use case or example to validation.
1. Clarify that validation is not error handling; link people to the [forms guidance](https://designsystem.digital.gov/components/form/#identifying-required-fields-2) for how to manage error states or requirements
2. Clarify that the validation component is used primarily (or exclusively) with the [text input component](https://designsystem.digital.gov/components/text-input/) (if that's not the case we need to share an example of how to use it with other form components)
3. Add a section for accessibility guidance. 
- [ ] Ensure that people use a checkmark that sufficiently passes color contrast. 
- [ ] ![image](https://github.com/uswds/uswds-site/assets/117778073/345bf628-2377-4c1a-bff2-c5e3831642a1)
The current example does pass for graphical objects.
- [ ] Ensure that screen reader users are able to hear if they have successfully entered the correct information/characters to achieve validation status
4. Recommend including a "When not to use validation" section. Do not use it for error handling; not recommended for top-of-page alerts.
5. Recommend adding placement guidance. If you have a text input field in the middle of a form do not put the validation feedback box at the top of a page or elsewhere. It can cause confusion for the user.
6. Recommend clarifying in the first section under "guidance" that Data validation means checking the data before performing operations that could fail. I'd suggest we consider calling it "data validation" if we are only going to allow people the option of using a text input component.

### Describe alternatives you've considered

The Blazorise design system has a lot of guidance on their validation page https://blazorise.com/docs/components/validation
Their code isn't fully accessible using assistive technology. However they give several examples of when and how to implement. They also give examples of inline validation which may be another issue we can consider in the future.

### Additional context

Happy to meet about this since I realize it's a lot of content suggestions. Validation is a slightly confusing component and may need to be discussed in a meeting where we screen share so everyone can hear how JAWS interprets the code.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.
